functions.js

```diff
@@ -0,0 +1,116 @@
+function upperCase(string){
+  return string.toUpperCase();
+}
+
+function lowerCase(string){
+  return string.toLowerCase();
+}
+
+function capitalize(string, all){
+  if (all){
+    return string.split(' ').map(function(a){
+      return capitalize(a, false);
+    }).join(' ');
+  } else {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+}
+
+function kebabCase(string){
+  return string.toLowerCase().replace(/ /g, '-');
```

Comment:
Consider replacing characters other than spaces as well, such as punctuation. `\W` (capital W) is a helpful [character class](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#character-classes) for that. If you do that, you'll also want to remove duplicate hyphens as well as leading/trailing ones.